Permanent/Common/Base/SwiftUIViews/GradientCustomBarView.swift

```diff
@@ -18,35 +19,93 @@ struct GradientProgressBarView: View {
                 HStack(spacing: 0) {
                     Text("\(value) ")
                         .textStyle(SmallXXXSemiBoldTextStyle())
-                        .foregroundColor(.white)
+                        .foregroundColor(colorScheme.textColor)
                     if value.isNotEmpty {
                         Text("used")
                             .textStyle(SmallXXXRegularTextStyle())
-                            .foregroundColor(.white)
+                            .foregroundColor(colorScheme.textSecondaryColor)
                     }
                 }
                 Spacer()
                 Text("\(maxValue)")
                     .textStyle(SmallXXXSemiBoldTextStyle())
-                    .foregroundColor(.white)
+                    .foregroundColor(colorScheme.textColor)
             }
             .padding(.horizontal)
-            ProgressView(value: sizeRatio)
-                .progressViewStyle(CustomBarProgressStyle(color: .white, height: 8, cornerRadius: 3))
-                .frame(height: 12)
-                .padding(.horizontal)
-                .id(redraw)
+            switch colorScheme {
+            case .settings:
+                ProgressView(value: sizeRatio)
+                    .progressViewStyle(CustomBarProgressGradientStyle(colorScheme: .settings, height: 8, cornerRadius: 3))
+                    .frame(height: 12)
+                    .padding(.horizontal)
+                    .id(redraw)
+            case .storage:
+                ProgressView(value: sizeRatio)
+                    .progressViewStyle(CustomBarProgressStyle(color: .white, height: 8, cornerRadius: 3))
+                    .frame(height: 12)
+                    .padding(.horizontal)
+                    .id(redraw)
+            }
         }
         .onChange(of: sizeRatio, perform: { newValue in
             updateRedraw()
         })
-        .frame(maxHeight: 72)
-        .background(Gradient.purpleYellowGradient)
+        .frame(maxHeight: colorScheme.frameHeightSize)
+        .background(colorScheme.backgroundColor)
         .cornerRadius(12)
-        .padding(16)
+        .padding(colorScheme == .storage ? 16  : 0)
     }
     
     func updateRedraw() {
         redraw = UUID()
     }
 }
+
+enum ColorSchemeForProgressBar {
+    case settings
```

Comment:
Instead of naming them for a specific page, you could do something like:
case white
case purpleYellow

Reply:
This should help in the future.